libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/DynamicBeginDialogSerializer.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Loaders;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Resources;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Actions
+{
+    /// <summary>
+    /// Internal serializer for DynamicBeginDialog which binds the x.dialog resourceId to the DyanmicBeginDialog.Dialog property.
+    /// </summary>
+    internal class DynamicBeginDialogSerializer : ICustomDeserializer
```

Comment:
I'd call this `DynamicBeginDialogDeserializer` since we use it to deserialize rather than serialize